### PR TITLE
[FLINK-6682] [checkpoints] Improve error message in case parallelism …

### PR DESCRIPTION
**Pull request by zhangminglei**
…exceeds maxParallelism

Thanks for contributing to Apache Flink. Before you open your pull request, please take the following check list into consideration.
If your changes take all of the items into account, feel free to open your pull request. For more information and/or questions please refer to the [How To Contribute guide](http://flink.apache.org/how-to-contribute.html).
In addition to going through the list, please provide a meaningful description of your changes.

- [ ] General
  - The pull request references the related JIRA issue ("[FLINK-XXX] Jira title text")
  - The pull request addresses only one issue
  - Each commit in the PR has a meaningful commit message (including the JIRA id)

- [ ] Documentation
  - Documentation has been added for new functionality
  - Old documentation affected by the pull request has been updated
  - JavaDoc for public methods has been added

- [ ] Tests & Build
  - Functionality added by the pull request is covered by tests
  - `mvn clean verify` has been executed successfully locally or a Travis build has passed

### CODE REVIEW

**Comment by zhangminglei** [comment]
Hi @zentol . Please helps review if you are free, should I add some extra information ? Thanks.

**Comment by zhangminglei** [comment]
Thank you so much @zentol . I did learn a lot from those. Also, the code have been updated. Please review. Thanks again.

**Comment by zentol** [comment]
Looks good to me, let me just quickly try it out.

File: flink-runtime/src/main/java/org/apache/flink/runtime/checkpoint/StateAssignmentOperation.java [hunk 1]
```diff
@@ -464,6 +464,14 @@ private void collectPartionableStates(
 	private static void checkParallelismPreconditions(OperatorState operatorState, ExecutionJobVertex executionJobVertex) {
 		//----------------------------------------max parallelism preconditions-------------------------------------
 
+		if (operatorState.getMaxParallelism() < executionJobVertex.getParallelism()) {
+			throw new IllegalStateException("The state for task " + executionJobVertex.getJobVertexId() +
+				" can not be restored. The maximum parallelism " + operatorState.getMaxParallelism() +
```
Comment (zentol) [inline flink-runtime/src/main/java/org/apache/flink/runtime/checkpoint/StateAssignmentOperation.java]: can you added braces around the (max)parallelism? The error message currently looks like this:

```
Caused by: java.lang.IllegalStateException: The state for task adf090656b210b1609ad3203d4ee7329 can not be restored. The maximum parallelism 128 of the restored state is lower than the configured parallelism 140. Please reduce the parallelism of the task to be lower or equal to the maximum parallelism.
```

But i think
```
Caused by: java.lang.IllegalStateException: The state for task adf090656b210b1609ad3203d4ee7329 can not be restored. The maximum parallelism (128) of the restored state is lower than the configured parallelism (140). Please reduce the parallelism of the task to be lower or equal to the maximum parallelism.
```
looks nicer.

Comment (zhangminglei) [inline flink-runtime/src/main/java/org/apache/flink/runtime/checkpoint/StateAssignmentOperation.java]: Yes. I have updated the code. Please helps to check. :XD

**Comment by zentol** [comment]
+1.

**Comment by zentol** [comment]
merging.

**Comment by zhangminglei** [comment]
Thanks for your review.

**Pull request closed without merging**